src/js/sniffer.js

```diff
@@ -26,6 +26,13 @@ const os = {
      **/
     get android() {
         return /android/i.test( ua );
+    },
+
+    /**ua
+     * @type {string}
+     **/
+    get safari() {
+        return /^((?!chrome|android).)*safari/i.test( ua );
```

Comment:
I don't understand the negative chrome|android lookahead. After reading the issue again, I realize you probably need to add an "os.macos" and a "browser.safari" sniffer right?

Reply:
We can't detect the safari browser by only check the string: `safari`  from `navigator.userAgent` .  because  `safari` exists in the userAgent for Chrome/Android, too. That's why I have to exceeded `chrome` and `android` here.
![image](https://user-images.githubusercontent.com/29697780/100702215-4518b180-33dc-11eb-8573-5d01bdb63437.png)

Reply:
Oh, right. Thanks